### DISCUSSION TITLE
brainstorm on pattern sharing

### DISCUSSION THREAD

**Discussion by felixroos** [tidalcycles/strudel#55] [Ideas]
It would be nice if the repl could do more in term of pattern sharing. just some ideas thrown into the room:

- github connector with the ability to save patterns as gist + to view gists/patterns from other users and fork them
- .. or another form of auth that is simpler, but let's people list / create / fork
- [x] builtin functionality to shorten the url, as long links will not work everywhere. either use some url shortening API or write own one that maps some short id to a base64 code string. now implemented with supabase
- twitter share link? (like hydra)
- musical code golfing?
- implement strudel in https://observablehq.com/ ??

**Comment by yaxu** [comment]
Related - a way to share samples alongside code. Dirt-samples are disorganised/uncleared. Clean-samples was an effort to rationalise things with a simple way of sharing cc licensed sample packs: https://github.com/tidalcycles/Clean-Samples
(see the issues there for discussion around metadata etc).

Or maybe this would be better as a separate discussion..

**Comment by jarmitage** [comment]
- iframe code generator (similar to YouTube or SoundCloud embed, etc)

**Reply by felixroos** [reply]
a first step: https://codesandbox.io/s/strudel-repl-component-2mxo8b?file=/index.html

**Reply by felixroos** [reply]
related, as mentioned in chat: discourse plugin that uses the web component

**Comment by jarmitage** [comment]
So far we have sharing Strudel URLs, sharing Strudel code, and embedding Strudel REPLs.

How about sharing Strudel code in-REPL, either through joining the same 'room' (Troop/Gibber), or by sending/receiving code across live Strudel sessions in a P2P way, or in a modular synth way like Hydra? For the latter you could have a type of pattern output that sends the code to another client and runs it there locally.

**Reply by felixroos** [reply]
related https://github.com/tidalcycles/strudel/issues/33 , modular synth way is also interesting, but would you then hear the end product of the sound or only up till your end?

**Comment by chaosprint** [comment]
I have a `share` button on https://glicol.web.app/
It's basically Firebase service and that's actually where I started.
My initial thought was to use it as a playground after I got the `.org` domain but as Google service is not available in some areas, it's nearly deprecated.
Still, the sharing mechanism is there and it's something I would miss with current decentralised solutions.

**Reply by felixroos** [reply]
also a possibility for strudel. maybe there are other services to look out for
